src/main/java/io/vertx/java/spi/cluster/impl/jgroups/services/DefaultRpcExecutorService.java

```diff
@@ -80,13 +79,16 @@ public DefaultRpcExecutorService(RpcDispatcher dispatcher) {
     try {
       NotifyingFuture<RspList<T>> notifyingFuture = this.<T>execute(action, options);
       notifyingFuture.setListener((future) -> {
-        try {
-          RspList<T> rspList = future.get();
-          T result = futureDone(rspList);
-          handler.handle(Future.succeededFuture(result));
-        } catch (Exception e) {
-          handler.handle(Future.failedFuture(e));
-        }
+        vertx.executeBlocking(
```

Comment:
To execute a callback on the correct thread you should use context.runOnContext(). executeBlocking is for running blocking code.